### ISSUE TITLE
Not working with Visual Basic Winforms project

### ISSUE THREAD

**Issue by shaxxx** [vescon/MethodBoundaryAspect.Fody#126]
Weaver is trowing exception in VB.net Winforms project.
I even tried small proof-of-concept project with no additional dependencies.

> Error		Fody: An unhandled exception occurred:
> Exception:
> Failed to execute weaver C:\Users\isako\.nuget\packages\methodboundaryaspect.fody\2.0.149\build\..\weaver\MethodBoundaryAspect.Fody.dll
> Type:
> System.Exception
> StackTrace:
>    at InnerWeaver.ExecuteWeavers() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 222
>    at InnerWeaver.Execute() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 113
> Source:
> FodyIsolated
> TargetSite:
> Void ExecuteWeavers()
> Object reference not set to an instance of an object.
> Type:
> System.NullReferenceException
> StackTrace:
>    at MethodBoundaryAspect.Fody.ModuleWeaver.IsMethodBoundaryAspect(TypeDefinition attributeTypeDefinition)
>    at System.Linq.Enumerable.WhereSelectEnumerableIterator`2.MoveNext()
>    at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
>    at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
>    at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
>    at MethodBoundaryAspect.Fody.ModuleWeaver.WeaveType(ModuleDefinition module, TypeDefinition type, Collection`1 assemblyMethodBoundaryAspects)
>    at MethodBoundaryAspect.Fody.ModuleWeaver.WeaveTypeAndNestedTypes(ModuleDefinition module, TypeDefinition type, Collection`1 assemblyMethodBoundaryAspects)
>    at MethodBoundaryAspect.Fody.ModuleWeaver.Execute(ModuleDefinition module)
>    at InnerWeaver.ExecuteWeavers() in C:\projects\fody\FodyIsolated\InnerWeaver.cs:line 186

I've spent half a day trying to debug the damn thing. 
Here's the problem

![1](https://github.com/user-attachments/assets/4e2278ea-393c-4934-a9d9-ab81bf90eab3)

Apparently, 

https://github.com/vescon/MethodBoundaryAspect.Fody/blob/1c94babb56aedcede20b1616f527576bac5a0579/src/MethodBoundaryAspect.Fody/ModuleWeaver.cs#L275

will call `IsMethodBoundaryAspect` with null value when AttributeType is `Microsoft.VisualBasic.CompilerServices.StandardModuleAttribute`
and the weaver will crash on the 
https://github.com/vescon/MethodBoundaryAspect.Fody/blob/1c94babb56aedcede20b1616f527576bac5a0579/src/MethodBoundaryAspect.Fody/ModuleWeaver.cs#L261

Now, when we know the cause, fix is easy
```

        private bool IsMethodBoundaryAspect(TypeDefinition attributeTypeDefinition)
        {
            var currentType = attributeTypeDefinition?.BaseType;
            if (currentType == null) return false;
            do
            {
                if (currentType.FullName == AttributeFullNames.OnMethodBoundaryAspect)
                    return true;

                currentType = currentType.Resolve().BaseType;
            } while (currentType != null);

            return false;
        }
```

This is pretty much urgent for me, so feedback on this would be nice.
Thanks.

**Comment by Ralf1108** [comment]
Hi @shaxxx,

as I understand you in your VB case there will be a null returned from "attributeTypeDefinition?.BaseType" and then we should just return false?

"if (currentType == null) return false;"

Is this the expected behavior or should the attribute changed specifically for VB.Net winforms code?

**Comment by Ralf1108** [comment]
fixed in version v2.0.150

**Comment by shaxxx** [comment]
I can confirm it's working.
